File: samples/java-app/src/main/java/com/appambit/javaapp/models/CmsExampleModel.java
package com.appambit.javaapp.models;

import com.appambit.sdk.utils.JsonKey;

import java.util.List;

public class CmsExampleModel {
    @JsonKey("id")
    public String id;
    @JsonKey("title")
    public String title;
    @JsonKey("body")
    public String body;
    @JsonKey("category")
    public List<String> category;
    @JsonKey("author")
    public String author;

    @JsonKey("featured_image_url")
    public String featuredImage;

    @JsonKey("likes")
    public int likes;

    @JsonKey("rating")
    public double rating;

    @JsonKey("reading_time")
    public int readingTime;

    @JsonKey("is_published")
    public Boolean isPublished;

    @JsonKey("published_at")
    public String publishedAt;
}

File: samples/kotlin-app/src/main/java/com/appambit/kotlinapp/models/CmsExampleModel.kt
package com.appambit.kotlinapp.models

import com.appambit.sdk.utils.JsonKey

class CmsExampleModel {
    @field:JsonKey("id")
    var id: String? = null
    @field:JsonKey("title")
    var title: String? = null
    @field:JsonKey("body")
    var body: String? = null
    @field:JsonKey("category")
    var category: List<String>? = null
    @field:JsonKey("author")
    var author: String? = null

    @JsonKey("featured_image_url")
    var featuredImage: String? = null

    @field:JsonKey("likes")
    var likes: Int = 0
    @field:JsonKey("rating")
    var rating: Double = 0.0

    @JsonKey("reading_time")
    var readingTime: Int = 0

    @JsonKey("is_published")
    var isPublished: Boolean? = null

    @JsonKey("published_at")
    var publishedAt: String? = null
}
